### ISSUE TITLE
utop: core dumped  after switch to ocaml 4.14.0

### ISSUE THREAD

**Issue by klartext** [ocaml/opam#5115]
Switched to Ocaml 4.14.0.

Then I wanted to use utop.
I got a core dump (see below).

`which utop`  was answered with
`/home/oliver/.opam/4.12.0/bin/utop`
even though the error message mentioned `4.14.0` stublibs.

Then installed utop to be matching to 4.14.0 switch.
Next call to utop got me the same error message:
```
<prompt>~$ utop
Fatal error: cannot load shared library dllthreads
Reason: /home/oliver/.opam/4.14.0/lib/ocaml/stublibs/dllthreads.so: undefined symbol: caml_stop_stack_overflow_detection
Aborted (core dumped)
```
What utop is in use now?
`/home/oliver/.opam/4.14.0/bin/utop`

-----

`opam config report` output:

```
# opam config report
# opam-version         2.1.2 
# self-upgrade         no
# system               arch=x86_64 os=linux os-distribution=arch os-version=unknown
# solver               builtin-mccs+glpk
# install-criteria     -removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed
# upgrade-criteria     -removed,-count[avoid-version,changed],-count[version-lag,solution],-count[missing-depexts,changed],-new
# jobs                 3
# repositories         1 (http) (default repo at b5838ec9)
# pinned               0
# current-switch       4.14.0
# ocaml:native         true
# ocaml:native-tools   true
# ocaml:native-dynlink true
# ocaml:stubsdir       /home/oliver/.opam/4.14.0/lib/ocaml/stublibs:/home/oliver/.opam/4.14.0/lib/ocaml
# ocaml:preinstalled   false
# ocaml:compiler       4.14.0
```

**Comment by dra27** [comment]
Did you run `eval $(opam env)` after switching to 4.14.0? What's the shebang of `/home/oliver/.opam/4.12.0/bin/utop` and `/home/oliver/.opam/4.14.0/bin/utop`?

**Comment by klartext** [comment]
I think I did it later (it was a bid late), but can't exclude the possibility I might have forgotten it, even though I think I did at least also test with new terminal (have a source command there in the .bashrc  to source `$HOME/.opam/opam-init/init.sh`).
Next day - computer was booted freshly, utop was working.
Could it be the case that the problem came from a different side, not the omam env thing?

It's now working.
